### PR TITLE
[ARCTIC-1650][AMS]: Optimize the serialization of table_optimizing_process.rewrite_input using gzip

### LINKED ISSUE #1650: [Improvement]: Reduce the size of the serialization of rewritten files
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

For tables with many rewritten files, it is more challenging for DB to read and write records in a single transaction.

### How should we improve?

Try to reduce the size of the serialization of rewritten files first

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by shidayang**
## Why are the changes needed?
fix #1650 and reduce the bytes when save record to table `table_optimizing_process `

## Brief change log
Add gzip in Object2ByteArrayConvert

## How was this patch tested?
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

File: ams/server/src/main/java/com/netease/arctic/server/utils/CompressUtil.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package com.netease.arctic.server.utils;
```
Comment (baiyangtx) [inline ams/server/src/main/java/com/netease/arctic/server/utils/CompressUtil.java:1]: Lack licences

File: ams/server/src/main/java/com/netease/arctic/server/persistence/converter/Object2ByteArrayConvert.java [hunk 2]
```diff
@@ -19,7 +20,7 @@ public void setParameter(PreparedStatement ps, int i, T parameter, JdbcType jdbc
       return;
     }
 
-    ps.setBytes(i, SerializationUtil.simpleSerialize(parameter).array());
+    ps.setBytes(i, CompressUtil.gzip(SerializationUtil.simpleSerialize(parameter).array()));
```
Comment (baiyangtx) [inline ams/server/src/main/java/com/netease/arctic/server/persistence/converter/Object2ByteArrayConvert.java:23]: Have you considered using Kryo? The size of Kryo is smaller then Java default serializer

**Review by baiyangtx — CHANGES_REQUESTED** [review]
You need add licences to the new added files.

File: ams/server/src/main/java/com/netease/arctic/server/utils/CompressUtil.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+public class CompressUtil {
+
+  public static byte[] gzip(byte[] bytes) {
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/utils/CompressUtil.java:29]: add javadoc for util method.

File: ams/server/src/test/java/com/netease/arctic/server/util/CompressUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package com.netease.arctic.server.util;
+
+import com.netease.arctic.server.utils.CompressUtil;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+
+public class CompressUtilTest {
```
Comment (zhoujinsong) [inline ams/server/src/test/java/com/netease/arctic/server/util/CompressUtilTest.java:9]: lack of licence.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Review by baiyangtx — APPROVED** [review]
LGTM

**Pull request merged**